src/ansi-c/expr2c.cpp

```diff
@@ -738,19 +738,18 @@ std::string expr2ct::convert_struct_type(
   {
     dest+=" {";
 
-    for(struct_typet::componentst::const_iterator
-        it=struct_type.components().begin();
-        it!=struct_type.components().end();
-        it++)
+    for(const struct_union_typet::componentt &component :
+      struct_type.components())
```

Comment:
I thought the general approach to `auto` was only when the type was either when the type was repeated on the RHS or for super obvious types like iterators? I figured this was neither but can change if you think it is better.